lib/routes/account.js

```diff
@@ -341,17 +362,33 @@ module.exports = function (
         var email = form.email
         var authPW = Buffer(form.authPW, 'hex')
         var service = request.payload.service || request.query.service
-        var emailRecord, sessionToken, device
+        var redirectTo = request.payload.redirectTo
+        var resume = request.payload.resume
+        var tokenVerificationId = crypto.randomBytes(16)
+        var emailRecord, sessionToken, device, keyFetchToken
+        var doSigninConfirmation = shouldEnableSigninConfirmation()
 
         customs.check(request.app.clientAddress, email, 'accountLogin')
           .then(readEmailRecord)
           .then(createSessionToken)
+          .then(createKeyFetchToken)
           .then(upsertDevice)
           .then(emitSyncLoginEvent)
-          .then(sendNewDeviceLoginNotification)
+          .then(sendVerifyLoginEmail)
           .then(createResponse)
           .done(reply, reply)
 
+        function shouldEnableSigninConfirmation() {
+          // Sign-in confirmation is determined by whether the feature is enabled and
+          // if it falls within config sample rate
+          if (config.signinConfirmation.enabled) {
+            var rand = Math.floor(Math.random() * (100 + 1))
+            return rand < (config.signinConfirmation.sample_rate * 100)
+          } else {
+            return false
+          }
+        }
```

Comment:
I wonder if we can make this value stable for a given user, so that they either always get the emails, or always dont.  One idea would be to interpret the first N bytes of their uid as an integer, and use that in place of that call to `Math.random` here.